js/api.ts

```diff
@@ -70,22 +70,18 @@ if (jsEnv) {
     // exported by Skydive that makes use of the REST client
     // so we get support for authentication
     makeRequest = function (client: Client, url: string, method: string, body: string, opts: Object) : any {
-        return new Promise(function (resolve, reject) {
-            var err: any
-            var data: any
-            try {
-                var output = request(url, method, body)
-                if (output && output.length > 0) {
-                    data = JSON.parse(output)
-                } else {
-                    data = null
-                }
-                resolve(data)
+        var data: any
+        try {
```

Comment:
nit: there is no need to try/catch if we just throw e on error

Reply:
removed